TicTacToe/tictactoe.py

```diff
@@ -0,0 +1,51 @@
+list_of_winners_to_ensure_only_one_player_won = []
+
+def column_check(player, column_number):
+    global list_of_winners_to_ensure_only_one_player_won
+    if game_board[0][column_number] == player and game_board[1][column_number] == player and game_board[2][column_number] == player:
+        list_of_winners_to_ensure_only_one_player_won.append(player)
+
+def row_check(player, row_number):
+    global list_of_winners_to_ensure_only_one_player_won
+    if game_board[row_number][0] == player and game_board[row_number][1] == player and game_board[row_number][2] == player:
+        list_of_winners_to_ensure_only_one_player_won.append(player)
+
+def diagonal_check(player):
+    global list_of_winners_to_ensure_only_one_player_won
+    if game_board[0][0] == player and game_board[1][1] == player and game_board[2][2] == player or game_board[2][0] == player and game_board[1][1] == player and game_board[0][2] == player:
+        list_of_winners_to_ensure_only_one_player_won.append(player)
+
+def __main__(input_matrix):
+    global game_board
+    global list_of_winners_to_ensure_only_one_player_won
+
+    game_board = input_matrix
+
+    # Has anyone completed a row?
+    for y in range(1,3):
```

Comment:
you assume here the number of players is always 2 and matrix size is always 3X3. 
it does work in the short term, but please come up with a solution that would also work for other matrix sizes.

this changes a-lot in the checking functions so I will only review them after this major change.